Scala/spectral-norm/spectralnorm_idiomatic.scala

```diff
@@ -0,0 +1,90 @@
+/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+   contributed by Isaac Gouy
+   parallel by the Anh Team
+   Scala Futures version by Robert Wilton
+ */
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.annotation.tailrec
+import ExecutionContext.Implicits.global
+
+object spectralnorm {
+  // Ordinary and transposed versions of infinite matrix
+  val A = (x: Int, y: Int) => 1.0 / ((x + y) * (x + y + 1) / 2 + x + 1)
+  val At = (x: Int, y: Int) => 1.0 / ((y + x) * (y + x + 1) / 2 + y + 1)
+
+  case class Ctx(n: Int) {
+    val u, v, tmp = Array.fill(n)(1.0)
+    private implicit val ctx: Ctx = this
+    val multiplyAv = multiply(A) _
+    val multiplyAtv = multiply(At) _
+  }
+
+  def main(args: Array[String]): Unit = {
+    val n = if (args.length > 0) args(0).toInt else 100
+    implicit val ctx: Ctx = Ctx(n)
+
+    // Calculate the chunks and perform calculation.
+    val chunks = {
+      val threads = Runtime.getRuntime.availableProcessors
+      val size = 1 + n / threads
+      for {
+        threadId <- 0 until threads
+        start = threadId * size
+        end = ((threadId + 1) * size).min(n)
+      } yield start until end
+    }
```

Comment:
I have the same remark here. We should just submit `n` jobs to the `ExecutionContext` and let it figure out how to efficiently execute them.